### PR TITLE
feat: Add support for Pennylane 0.29.1

### PR DESCRIPTION
**Pull request by math411**
*Issue #, if available:*

*Description of changes:*
Add support for Pennylane version 0.28.0

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ x ] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md) doc
- [ x ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [ x ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ x ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ x ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
     package_dir={"": "src"},
     install_requires=[
         "amazon-braket-sdk>=1.35.0",
-        "pennylane>=0.25.1,<=0.27.0",
+        "pennylane==0.28.0",
```
Comment (trbromley) [inline setup.py:31]: Note that PL version `0.29.0` was released this week :+1:

Comment (math411) [reply setup.py:31]: I shall be working on a follow up here to support this. There are some cases where an upgrade is needed.

Comment (trbromley) [reply setup.py:31]: As a follow up, the latest PL version is `0.29.1` due to an issue with `autoray`.

File: src/braket/pennylane_plugin/translation.py [hunk 2]
```diff
@@ -153,9 +153,8 @@ def translate_operation(
             parameters.append(new_param)
     else:
         parameters = [
-            p.numpy() if isinstance(p, qml.numpy.tensor) else p for p in operation.parameters
```
Comment (math411) [inline src/braket/pennylane_plugin/translation.py:156]: https://github.com/PennyLaneAI/pennylane/pull/3659/files#diff-ed5f9c0de17892f74e5119e2a032fe277eaa330ec4df0c249c7a75af7be6585e

File: setup.py [hunk 3]
```diff
@@ -28,9 +28,7 @@
     package_dir={"": "src"},
     install_requires=[
         "amazon-braket-sdk>=1.35.0",
-        "pennylane>=0.25.1,<=0.27.0",
-        # https://github.com/PennyLaneAI/pennylane/issues/3867
-        "autoray==0.6.0",
+        "pennylane==0.29.1",
```
Comment (math411) [inline setup.py:31]: Minimum version that fixes the autoray issue: https://github.com/PennyLaneAI/pennylane/issues/3867#issuecomment-1456756197

File: setup.py (hunk 1 above)
Comment (math411) [reply setup.py]: Updated to reflect this and the changes with `>=0.29.0`

**Review by speller26 — APPROVED** [review]

File: src/braket/pennylane_plugin/translation.py [hunk 4]
```diff
@@ -310,7 +309,7 @@ def _(_: qml.PhaseFlip, parameters):
 
 @_translate_operation.register
 def _(_: qml.QubitChannel, parameters):
-    K_list = [np.asarray(matrix) for matrix in parameters[0]]
+    K_list = [np.asarray(matrix) for matrix in parameters]
```
Comment (math411) [inline src/braket/pennylane_plugin/translation.py:312]: https://github.com/PennyLaneAI/pennylane/pull/3659/files#diff-ed5f9c0de17892f74e5119e2a032fe277eaa330ec4df0c249c7a75af7be6585e

**Pull request merged**